src/test/java/uk/gov/hmcts/ccd/data/definition/DefaultCaseDefinitionRepositoryTest.java

```diff
@@ -64,9 +74,23 @@ public void shouldThrowResourceNotFoundExceptionWhenGetBaseTypesIsCalledAndResou
     @Test
     public void shouldFailToGetClassificationForUserRole() {
         HttpClientErrorException httpException = new HttpClientErrorException(HttpStatus.INTERNAL_SERVER_ERROR);
-        doThrow(httpException).when(restTemplate).exchange(anyString(), any(), any(), any(Class.class));
+        doThrow(httpException).when(restTemplate).exchange(anyString(), any(HttpMethod.class), any(), any(Class.class));
 
         ServiceException exception = assertThrows(ServiceException.class, () -> caseDefinitionRepository.getUserRoleClassifications("nor_defined"));
         assertThat(exception.getMessage(), startsWith("Error while retrieving classification for user role nor_defined because of "));
     }
+
+    @Test
+    public void test_propagation_called_once_and_only_once() {
+        CaseType caseType = new CaseType();
+        int fieldCount = 1 + (int)(Math.random() * 20);
```

Comment:
What is the point of adding a random number of fields? Lets call twice fields.add(Mockito.mock(CaseField.class)); and it is much simpler. If we want to test one and many we should create two separate tests.